ang/civicase/ActivityCard.js

```diff
@@ -1,56 +1,103 @@
-(function(angular, $, _) {
+(function (angular, $, _) {
+  var module = angular.module('civicase');
 
-  function activityCard($scope, getActivityFeedUrl, dialogService, templateExists) {
+  module.directive('caseActivityCard', function () {
+    return {
+      restrict: 'A',
+      templateUrl: '~/civicase/ActivityCard.html',
+      controller: caseActivityCardController,
+      scope: {
+        activity: '=caseActivityCard',
+        refresh: '=refreshCallback',
+        editActivityUrl: '=?editActivityUrl'
+      }
+    };
+  });
+
+  function caseActivityCardController ($scope, getActivityFeedUrl, dialogService, templateExists, crmApi) {
     var ts = $scope.ts = CRM.ts('civicase');
-    $scope.CRM = CRM;
     $scope.activityFeedUrl = getActivityFeedUrl;
+    $scope.CRM = CRM;
     $scope.templateExists = templateExists;
 
-    $scope.isActivityEditable = function(activity) {
+    // console.log($scope.activity.category);
+
+    $scope.isActivityEditable = function (activity) {
```

Comment:
Should we have a documentation for this function too. The function name seems self explanable however. What do you think ?

Reply:
We should have documentation for every function because we not only document its purpose, but we also document the parameters and return type. In this case I have left it out, because I have not done anything related to this functionality. This is used in the Activity menu, which is not yet implemented. So we would add it when we work on this functionality itself.